src/wazuh_db/wdb_syscollector.c

```diff
@@ -833,9 +855,15 @@ int wdb_port_save(wdb_t * wdb, const char * scan_id, const char * scan_time, con
 int wdb_port_insert(wdb_t * wdb, const char * scan_id, const char * scan_time, const char * protocol, const char * local_ip, int local_port, const char * remote_ip, int remote_port, int tx_queue, int rx_queue, long long inode, const char * state, int pid, const char * process, const char * checksum, const char * item_id, const bool replace) {
     sqlite3_stmt *stmt = NULL;
 
+    if (NULL == protocol || NULL == local_ip || local_port < 0 || inode < 0) {
+        if(checksum && 0 != strcmp(SYSCOLLECTOR_LEGACY_CHECKSUM_VALUE, checksum)) {
+            wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_PORTS, item_id);
+        }
```

Comment:
```suggestion
        wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_PORTS, item_id);
```
the item_id is a sync field, no need to re-identify it, check the call tree. (For legacy code, item_id is null)

Reply:
Idem